sim-cli/src/main.rs

```diff
@@ -199,13 +199,15 @@ async fn main() -> anyhow::Result<()> {
         None
     };
 
+    let (shutdown_trigger, shutdown_listener) = triggered::trigger();
     let sim = Simulation::new(
         clients,
         validated_activities,
         cli.total_time,
         cli.expected_pmt_amt,
         cli.capacity_multiplier,
         write_results,
+        (shutdown_trigger, shutdown_listener),
```

Comment:
Oh very nice, I think I'm going to leave this for a follow up because I'll drop the commit where we actually use the simulator before merge (to be followed up with surfacing the config options to use it).